includes/blocks/course-theme/class-lesson-actions.php

```diff
@@ -0,0 +1,133 @@
+		$permalink = esc_url( get_permalink() );
+		$text      = esc_html( __( 'Complete lesson', 'sensei-lms' ) );
+
+		return ( '
+			<form class="sensei-course-theme-lesson-actions__complete-lesson-form" method="POST" action="' . $permalink . '">
+				' . $nonce . '
+				<input type="hidden" name="quiz_action" value="lesson-complete" />
+				<button type="submit" class="sensei-course-theme__button ' . $button_class . '" ' . $disabled_attribute . '>
+					' . $text . '
+				</button>
+			</form>
+		' );
+	}
+
+	/**
+	 * Renders take quiz button.
+	 *
+	 * @param string $quiz_permalink Quiz permalink.
+	 * @param bool   $is_disabled    Whether it is disabled.
+	 *
+	 * @return string The take quiz button.
+	 */
+	private function render_take_quiz( string $quiz_permalink, bool $is_disabled ) : string {
+		$disabled       = $is_disabled ? 'aria-disabled="true"' : '';
+		$quiz_permalink = esc_url( $quiz_permalink );
+		$text           = esc_html__( 'Take quiz', 'sensei-lms' );
+
+		return '<a href="' . $quiz_permalink . '" class="sensei-course-theme__button is-primary" ' . $disabled . '>'
+			. $text .
+		'</a>';
+	}
+
+	/**
+	 * Renders the block.
+	 *
+	 * @param array $attributes The block attributes.
+	 *
+	 * @access private
+	 *
+	 * @return string The block HTML.
+	 */
+	public function render( array $attributes = [] ) : string {
+		$lesson_id = Sensei_Utils::get_current_lesson();
+		$user_id   = get_current_user_id();
+
+		if ( empty( $lesson_id ) || empty( $user_id ) ) {
+			return '';
+		}
+
+		$course_id = Sensei()->lesson->get_course_id( $lesson_id );
+
+		if (
+			! Sensei_Course::is_user_enrolled( $course_id )
+			|| Sensei_Utils::user_completed_lesson( $lesson_id )
+		) {
+			return '';
+		}
+
+		$has_incomplete_prerequisite = ! Sensei_Lesson::is_prerequisite_complete( $lesson_id, $user_id );
+		$quiz_permalink              = Sensei()->lesson->get_quiz_permalink( $lesson_id );
+		$is_quiz_submitted           = Sensei()->lesson->is_quiz_submitted( $lesson_id, $user_id );
+		$is_pass_required            = Sensei()->lesson->lesson_has_quiz_with_questions_and_pass_required( $lesson_id );
+		$actions                     = [];
+
+		// Quiz button.
+		if ( ! empty( $quiz_permalink ) && ! Sensei()->lesson->is_quiz_submitted( $lesson_id, $user_id ) ) {
+			$take_quiz_button = $this->render_take_quiz( $quiz_permalink, $has_incomplete_prerequisite );
+			$actions[]        = '<li>' . $take_quiz_button . '</li>';
+		}
+
+		// Complete button.
+		if ( ! $is_pass_required ) {
+			$complete_button_class  = isset( $take_quiz_button ) ? 'is-secondary' : 'is-primary';
+			$complete_lesson_button = $this->render_complete_lesson( $complete_button_class, $has_incomplete_prerequisite );
+			$actions[]              = '<li>' . $complete_lesson_button . '</li>';
+		}
+
+		if ( empty( $actions ) ) {
+			return '';
+		}
+
+		return '<ul class="sensei-course-theme-lesson-actions">
```

Comment:
Not sure yet if we're better of not wrapping the buttons in a container or letting the header container control spacing & stuff, but we can tune it when it comes up (for the mobile layout probably)

Reply:
I think it's usually easier to control spacings around the elements in the parent, so the spacing can change depending on the context it is, but yeah, we can tweak later if needed! =)